### ISSUE TITLE
Can't compile Erlang OTP27 RC3  on ppc32 and ppc.spe because of zlib crash.

### ISSUE THREAD

**Issue by sailingareus** [erlang/otp#8474]
**Describe the bug**
We (Cisco Tail-f) are trying to compile ppc32 (powerpc) and ppc.spe and it cashes with following error when compiling leex.erl on vanilla OTP27 RC3 i.e. we have no added patches to OTP27 so it is the plain vanilla release.

```
=== Entering application parsetools
make[3]: Entering directory '/srv/confd-test/morty-ppc32-b-confd-test/confd/otp/lib/parsetools/src'
 ERLC   ../ebin/leex.beam
System process <0.0.0> terminated: {{case_clause,{error,{bad_lib,"Function not found zlib:close_nif/1"}}},[{zlib,on_load,0,[]},{erl_init,start,0,[]}]}
(no logger present) unexpected logger message: {log,error,"Error in process ~p with exit value:~n~p~n",[<0.0.0>,{{case_clause,{error,{bad_lib,"Function not found zlib:close_nif/1"}}},[{zlib,on_load,0,[]},{erl_init,start,0,[]}]}],#{error_logger=>#{emulator=>true,tag=>error},pid=><0.0.0>,time=>1715595168009021,gl=><0.0.0>}}
```
**To Reproduce**
 ```
 git clone -b OTP-27.0-rc3  https://github.com/erlang/otp.git
./configure --disable-year2038
make all
```
**Expected behavior**
That is possible to compile OTP RC3 on ppc32 and ppc.spe

**Affected versions**
OTP27 RC1 -R3.
We have tested to compile towards OTP-26.2.5 and it works fine 

**Additional context**
The build host is the following (Yocto Morty):
```
test@morty-ppc32-b:/confd-test/confd/otp$ uname -a   
Linux morty-ppc32-b 4.1.8-rt8+gbd51baf #1 SMP Fri Aug 4 17:06:00 UTC 2023 ppc ppc ppc GNU/Linux
```

I have included the erl_crash.dump
[erl_crash_dump.txt](https://github.com/erlang/otp/files/15294291/erl_crash_dump.txt)

NOTE we have also tried to compile natively on Linux x86_64. i686, aarch64, arm.hf and arm32. Plus Darwin Arm64 and x86_64 they work all fine. Cross compelling  ppc32,ppc.spe  and all of the above linux architectures works also fine. So it is only native build of ppc32 and ppc.spe that is failing.

We have tried to do a manual git bisect but it did not help very much because it got us to a merge commit that did not say much because it pointed to an other maintenance branch. So any suggestion how to help you debug this more would be help full:

_Failing commit:_
```
commit 1be425c7bce5b98e44c9660fe2b5d28f3c6f63d4
Merge: 0c5bbb9 2770d6f
Author: Lukas Larsson lukas@erlang.org
Date:   Thu Oct 12 10:05:04 2023 +0200

    Merge branch 'maint' into master
    
    * maint:
      Updated OTP version
      Prepare release
```

_Working commit:_
```
commit 2770d6f858aec6c044465cb123553f8649d18802
Merge: c9825a759c c41d424db4
Author:     Lukas Larsson lukas@erlang.org
AuthorDate: Thu Oct 12 10:04:59 2023 +0200
Commit:     Lukas Larsson lukas@erlang.org
CommitDate: Thu Oct 12 10:05:00 2023 +0200

    Merge branch 'maint-26' into maint
    
    * maint-26:
      Updated OTP version
      Prepare release OK

**Comment by garazdawi** [comment]
The working commit you have there is a commit on the 26 track, so it is not surprising that it works. You need to `git bisect` with `--first-parent` and then it will only try to run commits that are on the 27 track.

**Comment by sailingareus** [comment]
We have been doing some more investigations and and with help of my co-worker Alexandre we have figured out it is the same problem as https://github.com/erlang/otp/issues/8410. We have cherry-picked the commit and tested it and it is working.

**Comment by garazdawi** [comment]
Makes sense as they are both big-endian architecture problems.

It would be great if you could test the latest `master` commit as well, just to make sure that we haven't introduced something else after RC3 that breaks big-endian. We don't have any such machines to test on, so a bug or two are bound to slip through.

**Comment by sailingareus** [comment]
@garazdawi I have tested to build on master for  powerpc and powerpc.se and it worked fine. We have tested to build (and cross compile) and test OTP27 RC3 natively on Linux PPC32, PPC.SPE (32), arm32,. arm.hf (32) ,aarch64,  i686, x86_64. Build and tests have also been done towards Darwin arm64 and x86_64 on OTP27 RC3 and things look fine. Note so far tough we have not finished testing all powerPC big endian completely yet because  of limited recourses but the initial looks ok.
